doc/subsystems/networking/usbnet_setup.rst

```diff
@@ -0,0 +1,79 @@
+.. _usb_device_networking_setup:
+
+USB Device Networking
+#####################
+
+This page describes how to set up networking between a Linux host
+and a Zephyr application running on USB supported devices. A sample
+application from Zephyr source distribution is run on one of
+supported boards. The board is connected to Linux host using USB cable
+and provides an Ethernet interface to the host.
+
+Basic Setup
+***********
+
+To communicate with Zephyr application over newly created Ethernet
+interface we need to assign IP addresses and set routing table.
+
+After plugging USB cable to Linux host cdc_ether driver registers a new
+Ethernet device with provided MAC
+
+.. code-block:: console
+
+   cdc_ether 1-2.7:1.0 eth0: register 'cdc_ether' at usb-0000:00:01.2-2.7, CDC Ethernet Device, 00:00:5e:00:53:01
+
+If interface is not UP we need to set it UP and assign IP addresses.
+
+Choosing IP addresses
+=====================
+
+It make sense to choose addresses in the same subnet we have in Zephyr
+application. IP addresses usually set in the project configuration files
+and may be checked also from the shell with:
+
+.. code-block:: console
+
+   shell> net iface
+
+   Interface 0xa800e580 (Ethernet)
+   ===============================
+   Link addr : 00:00:5E:00:53:00
+   MTU       : 1500
+   IPv6 unicast addresses (max 2):
+           fe80::200:5eff:fe00:5300 autoconf preferred infinite
+           2001:db8::1 manual preferred infinite
+   ...
+   IPv4 unicast addresses (max 1):
+           192.0.2.1 manual preferred infinite
+
+Setting IPv4 address and routing
+================================
+
+.. code-block:: console
+
+   # ip address add dev enx00005e005301 192.0.2.2
+   # ip link set enx00005e005301 up
+   # ip route add 192.0.2.0/24 dev enx00005e005301
+
+Setting IPv6 address and routing
+================================
+
+.. code-block:: console
+
+   # ip address add dev enx00005e005301 2001:db8::2
+   # ip link set enx00005e005301 up
+   # ip -6 route add 2001:db8::/64 dev enx00005e005301
+
+Testing connection
+******************
+
+From the host we can test connection pinging Zephyr IP address with:
```

Comment:
change to "...test the connection by pinging the IP address of the board running Zephyr with:"